File: app/src/main/java/org/xdty/callerinfo/fragment/PreferenceActions.java
package org.xdty.callerinfo.fragment;

import android.preference.Preference;

public interface PreferenceActions {
    Preference findPreference(CharSequence key);

    void setChecked(int key, boolean checked);

    void onConfirmed(int key);

    void onConfirmCanceled(int key);

    void checkOfflineData();

    void resetOfflineDataUpgradeWorker();

    void addPreference(int parent, int child);

    void removePreference(int parent, int child);

    int getKeyId(String key);
}

File: app/src/main/java/org/xdty/callerinfo/settings/PreferenceActions.kt
package org.xdty.callerinfo.settings

import android.preference.Preference

interface PreferenceActions {
    fun findPreference(key: CharSequence?): Preference?
    fun setChecked(key: Int, checked: Boolean)
    fun onConfirmed(key: Int)
    fun onConfirmCanceled(key: Int)
    fun checkOfflineData()
    fun resetOfflineDataUpgradeWorker()
    fun addPreference(parent: Int, child: Int)
    fun removePreference(parent: Int, child: Int)
    fun getKeyId(key: String?): Int
}